performance_manager/lib/rt_trip_updates.py

```diff
@@ -155,18 +160,115 @@ def get_and_unwrap_tu_dataframe(
     events["is_moving"] = False
     events["pk_id"] = None
 
+    return events
+
+
+def join_gtfs_static(
+    trip_updates: pandas.DataFrame, db_manager: DatabaseManager
+) -> pandas.DataFrame:
+    """
+    join trip update dataframe to gtfs static records
+
+    adds "fk_static_timestamp" and "parent_station" columns to dataframe
+    """
+    # get unique "start_date" values from trip update dataframe
+    # with associated minimum "timestamp_start"
+    date_groups = trip_updates.groupby(by="start_date")["timestamp_start"].min()
+
+    # pylint: disable=duplicate-code
+    # TODO(ryan): the following code is duplicated in rt_vehicle_positions.py
+
+    # dictionary used to match minimum "timestamp_start" values to
+    # "timestamp" from StaticFeedInfo table, to be used as foreign key
+    timestamp_lookup = {}
+    for (date, min_timestamp) in date_groups.iteritems():
+        date = int(date)
+        min_timestamp = int(min_timestamp)
+        # "start_date" from trip updates must be between "feed_start_date"
+        # and "feed_end_date" in StaticFeedInfo
+        # minimum "timestamp_start" must also be less than "timestamp" from
+        # StaticFeedInfo, order by "timestamp" descending and limit to 1 result
+        # this should deal with multiple static schedules with possible
+        # overlapping times of applicability
```

Comment:
is there an ordering that we need to do to ensure the selection at 1 is the one we want?

Reply:
Yes, the ordering is done in the `sa.select` statement with `.order_by(StaticFeedInfo.timestamp.desc())` 

There is a bit of a race condition here where we need to have processed the matching GTFS Static data before any RT data processing.